src/perf/isolates.md

```diff
@@ -0,0 +1,347 @@
+When you spawn an isolate in Dart,
+it can process the work concurrently with the main isolate,
+without blocking it.
+
+You can read more about how isolates
+and the event loop work in Dart on
+the [concurrency page][] of the Dart
+documentation.
+
+[concurrency page]: {{site.dart-site}}/language/concurrency
+
+<iframe width="560" height="315" src="{{site.youtube-site}}/embed/vl_AaCgudcY" 
+ frameborder="0" allow="accelerometer; autoplay;encrypted-media; gyroscope; picture-in-picture" 
+ allowfullscreen></iframe>
+
+## Common use cases for isolates
+
+There is only one hard rule for when you should use isolates,
+and that's when large computations are causing your Flutter application
+to experience UI jank.
+This jank happens when there is any computation that takes longer than
+Flutter's frame gap.
+
+![Event jank diagram]({{site.url}}/assets/images/docs/development/concurrency/event-jank.png){:width="50%"}
+
+Any process _could_ take longer to complete,
+depending on the implementation
+and the input data,
+making it impossible to create an exhaustive list of
+when you need to consider using isolates.
+
+That said, isolates are commonly used for the following:
+
+- Reading data from a local database
+- Sending push notifications
+- Parsing and decoding large data files
+- Processing or compressing photos, audio files, and video files
+- Converting audio and video files
+- When you need asynchronous support while using FFI
+- Applying filtering to complex lists or filesystems
+
+## Message passing between isolates
+
+Dart's isolates are an implementation of the [Actor model][].
+They can only communicate with each other by message passing.
+When messages are "passed" between each other,
+they are generally copied from the sending isolate to the
+receiving isolate.
+This means that any value passed to an isolate,
+even if mutated on that isolate,
+doesn't change the value on the original isolate.
+
+The only objects that aren't copied when passed to an isolate
+are immutable objects that can't be changed anyway,
+such a String or an unmodifiable byte.
+When you pass an immutable object between isolates,
+it is transferred, rather than copied,
+for better performance.
+Because immutable objects can't be updated,
+this effectively retains the actor model behavior.
+
+An exception to this rule is
+when an isolate exits when it sends a message using the `Isolate.exit` method.
+Because the sending isolate won't exist after sending the message,
+it can pass the reference to the message while
+ensuring that only one isolate can access the message.
+
+The two lowest-level primitives that send messages are `SendPort.send`,
+which makes a copy of a mutable message as it sends,
+and `Isolate.exit`,
+which sends the reference to the message.
+Both `Isolate.run` and `compute`
+use `Isolate.exit` under the hood.
+
+## Short-lived isolates
+
+The easiest way to move a process to an isolate in Flutter is with
```

Comment:
It may be worth mentioning the overhead associated copying messages and spawning isolates so people get some intuition about the tradeoffs.